flexmeasures/data/models/validation_utils.py

```diff
@@ -59,3 +68,38 @@ def check_required_attributes(
         raise WrongTypeAttributeException(
             f"Sensor attributes are not of the required type:\n {error_message}"
         )
+
+
+def can_modify_role(current_user: "User", roles_to_modify: list["Role"]):  # noqa: F821
+    """Check if the current user can modify the roles.
+
+    :param current_user: The current user.
+    :param roles_to_modify: A list of roles to modify.
+    :return: True if the user can modify the roles, False otherwise.
+
+    The roles are:
+    - admin: can only be changed in CLI / directly in the DB
+    - admin-reader: can be added and removed by admins
+    - account-admin: can be added and removed by admins and consultants
+    - consultant: can be added and removed by admins and account-admins
+
+    """
+    for role in roles_to_modify:
```

Comment:
By this logic, I only need to be able to modify one of the roles in the list, and the function already returns `True`...

Only return `True` if all roles in `roles_to_modify` passed the test.